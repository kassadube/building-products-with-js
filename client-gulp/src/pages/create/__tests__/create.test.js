/* eslint-disable react/react-in-jsx-scope */

import configureMockStore from 'redux-mock-store';
import CreateWarraper, {Create} from '../index';

// create mock store
const mockStore = configureMockStore();

const question = {
  id: '0',
  text: 'test',
  expirationDate: new Date(2016, 1, 1, 1, 1, 1, 1), //'2017-05-07T10:22:37.364Z',
};


const user = {
  id: '0',
  login: 'test',
  registrationDate: '2017-05-07T10:22:37.364Z',
};

test('# Create Warraper', () => {
  const store = mockStore({auth: user});
  const component = (<CreateWarraper user={user} store={store} />);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});


test('# Create', () => {
  const component = (<Create />);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});

test('# doAddQuestion', () => {
  const newText = 'newTest';
  const newExpirationDate = new Date(2016, 1, 1, 1, 1, 1, 1);
  const doAddQuestion = ({text, expirationDate}) => {
    expect(text).toBe(newText);
    expect(expirationDate).toBe(newExpirationDate.toISOString());
  };
  const component = (
    <Create user={user} doAddQuestion={doAddQuestion} />
  );
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  const app = mount(component);
  // set new question text
  app.find('#questionText').getDOMNode().value = newText;
  // set new question expiration date
  app.find('#expirationDate').getDOMNode().value = newExpirationDate.toISOString();
  // click answer button
  app.find('button').simulate('click');

});
