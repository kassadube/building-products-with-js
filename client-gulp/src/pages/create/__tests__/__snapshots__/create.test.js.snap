// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Create 1`] = `
<div>
  <Component
    user={Object {}}
  />
  <div>
    <form
      className="form"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="questionText"
        >
          Question Text
        </label>
        <input
          className="form-control"
          id="questionText"
          placeholder="place your text..."
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="expirationDate"
        >
          Expiration Date
        </label>
        <input
          className="form-control"
          id="expirationDate"
          type="date"
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-default"
          id="createBTN"
          onClick={[Function]}
          type="submit"
        >
          Create new question
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`# Create Warraper 1`] = `
<Create
  doAddQuestion={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  user={Object {}}
/>
`;

exports[`# doAddQuestion 1`] = `
<div>
  <Component
    user={
      Object {
        "id": "0",
        "login": "test",
        "registrationDate": "2017-05-07T10:22:37.364Z",
      }
    }
  />
  <div>
    <form
      className="form"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="questionText"
        >
          Question Text
        </label>
        <input
          className="form-control"
          id="questionText"
          placeholder="place your text..."
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="expirationDate"
        >
          Expiration Date
        </label>
        <input
          className="form-control"
          id="expirationDate"
          type="date"
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-default"
          id="createBTN"
          onClick={[Function]}
          type="submit"
        >
          Create new question
        </button>
      </div>
    </form>
  </div>
</div>
`;
